src/rpc/blockchain.cpp

```diff
@@ -1232,6 +1243,57 @@ UniValue rollbackchain(const UniValue &params, bool fHelp)
     return NullUniValue;
 }
 
+UniValue reconsidermostworkchain(const UniValue &params, bool fHelp)
+{
+    if (fHelp || params.size() > 1)
+        throw runtime_error("reconsidermostworkchain \"[override]\"\n"
+                            "\nWill rollback the chain if needed and then sync to the most work chain.\n"
+                            "\nArguments:\n"
+                            "1. override      (boolean, optional, default=false)"
+                            "\nResult:\n"
+                            "\nExamples:\n" +
+                            HelpExampleCli("reconsidermostworkchain", "") +
+                            HelpExampleCli("reconsidermostworkchain", "\"true\"") +
+                            HelpExampleRpc("reconsidermostworkchain", "\"true\""));
+
+    // Find pindex of most work chain regardless of whether is is valid or not.
+    LOCK(cs_main);
+
+    // Get the set of chaintips
+    std::set<CBlockIndex *, CompareBlocksByHeight> setTips;
+    setTips = GetChainTips();
+
+    // Find the longest chaintip regardless if it is currently the active one.
+    CBlockIndex *pMostWork = chainActive.Tip();
+    for (CBlockIndex *pTip : setTips)
+    {
+        if (pMostWork->nChainWork < pTip->nChainWork)
+            pMostWork = pTip;
+    }
+
+    // Find where chainActive meets the most work chaintip
+    const CBlockIndex *pFork;
+    pFork = chainActive.FindFork(pMostWork);
+
+    // Rollback to the common forkheight so that both chains will be invalidated.
+    UniValue obj(UniValue::VARR);
+    obj.push_back(pFork->nHeight);
+    if (params.size() > 0)
+    {
+        // Set the rollbackchain override flag if there was one provided.
+        obj.push_back(params[0]);
+    }
+    rollbackchain(obj, false);
```

Comment:
I don't like that we are calling an RPC call directly.  But since this is in the context of a different RPC call, and does not process return data, ok.